OutOfSchool/OutOfSchool.WebApi.Tests/Services/ParentServiceTests.cs

```diff
@@ -159,6 +164,41 @@ public void BlockUnblockParent_WhenBlockUnblockParentDtoIsNull_ShouldThrowArgume
         Assert.ThrowsAsync<ArgumentNullException>(async () => await parentService.BlockUnblockParent(parentBlockUnblockValid));
     }
     #endregion
+
+    #region Delete
+    [Test]
+    public async Task Delete_WhenIdIsNotValid_ThrowException()
+    {
+        // Arrange
+        Guid parentId = Guid.NewGuid();
+        currentUserServiceMock.Setup(x => x.UserHasRights(It.IsAny<ParentRights>())).Returns(() => Task.FromResult(true));
+        parentRepositoryMock.Setup(r => r.GetById(parentId)).ReturnsAsync(null as Parent);
+
+        // Act and Assert
+        Assert.ThrowsAsync<ArgumentException>(async () => await parentService.Delete(parentId).ConfigureAwait(false));
+    }
+
+    [Test]
+    public async Task Delete_WhenIdIsValid_ShouldCallDeleteMethods()
+    {
+        // Arrange
+        Guid parentId = Guid.NewGuid();
+        Parent parent = ParentGenerator.Generate();
+        parent.Id = parentId;
```

Comment:
(Excessive code) As `ParentGenerator` already creates `Id` as `Guid.NewGuid()`, you don't need to do it additionally.
So it could be as simple as 
```
        Parent parent = ParentGenerator.Generate();
        var parentId = parent.Id;
```